### DISCUSSION TITLE
توضیحات درمورد بعضی نسخه‌ها

### DISCUSSION THREAD

**Discussion by SCR-IR** [SCR-IR/gnome-shamsi-calendar#15] [اخبار]
<div dir=rtl>
تا الآن سعی کردیم سازگاری و پشتیبانی از اکثر نسخه‌های قدیمی گنوم را همچنان حفظ کنیم؛ امّا متأسفانه به‌دلیل ایراد گرفتن سایت 
extensions.gnome.org
 مجبور خواهیم شد در نسخه‌های بعدی افزونه، پشتیبانی از گنوم‌های قدیمی‌تر را حذف کنیم.
<br/>
البته تلاش می‌کنیم که همیشه از نسخه‌ی گنوم موجود در آخرین توزیع پایدار «دبیان» پشتیبانی شود.
<br/>
لذا از پشتیبانی نسخه‌های قدیمی‌تر گنوم، معذوریم. البتّه نصب نسخه‌های قبلی افزونه در گنوم‌های قدیمی همچنان امکانپذیر خواهد بود.
</div>

**Comment by SCR-IR** [comment]
<div dir=rtl>
<b>نسخه‌های ناپایدار گنوم یا توزیع‌های لینوکس</b>
<br/><br/>
هیچ تضمینی برای سازگاری این افزونه با نسخه‌های ناپایدار توزیع‌های لینوکس وجود ندارد.
<br/>
به‌عنوان مثال، ممکن از در نسخه‌های بتای فدورا، اجرای افزونه با خطا همراه باشد. پس بهتر است برای نصب افزونه، فقط از نسخه‌های پایدار رسمی استفاده نمایید.
<br/>
تجربه بارها نشان داده که در نسخه‌های آزمایشی (غیررسمی) فدورا، اشکالات فراوانی وجود دارد که سرانجام در نسخه‌ی پایدار فدورا، برطرف شده‌اند. لذا بهتر است وقت خود را برای سازگارکردن افزونه با نسخه‌های آزمایشی فدورا هدر ندهیم!
</div>

**Comment by SCR-IR** [comment]
<div dir=rtl>
<b>درمورد نسخه ۱۸</b>
<br/><br/>
اکنون سازگاری با گنوم ۳.۳۶ تا ۴۲ وجود دارد. (۴۲ آزمایشی)
<br/>
احتمالاً اندکی پس از انتشار اوبونتو ۲۲.۰۴ ، پشتیبانی از گنوم ۳.۳۶ را متوقّف کنیم؛ امّا پشتیبانی از گنوم ۳.۳۸ تا زمان انتشار دبیان ۱۲ همچنان ادامه خواهد داشت.
<br/>
ملاک پشتیبانی از نسخه‌های گنوم، توزیع‌های مهم لینوکس هستند. ازجمله: دبیان، اوبونتو و غیره. البته فدورا قدری عجول بوده و اغلب، گنوم آن جدید است.
</div>

**Comment by SCR-IR** [comment]
<div dir=rtl>
<b>درمورد نسخه ۱۹</b>
<br/><br/>
سازگاری با گنوم ۴۲ تکمیل شد.
<br/>
در ۲ سال آینده به‌تدریج، پشتیبانی از گنوم ۳.۳۶ ، سپس ۳.۳۸ و درنهایت ۴۰ ، ۴۱ و GTK3 را متوقّف خواهیم کرد؛ البته نسخه‌های قبلی افزونه همچنان در سایت گنوم در دسترس هستند و این عدم پشتیبانی فقط برای نسخه‌های جدید اِعمال می‌شود.
<br/>
توصیه می‌کنیم همواره از جدیدترین نسخه‌های LTS و توزیع‌های نسبتاً پایدار استفاده بفرمایید. نصب توزیع‌هایی که زودبه‌زود تغییرات اساسی دارند (مثل فدورا) و یا نسخه‌های موقّت (مثل اوبونتو 22.09) و امثال اینها را هرگز توصیه نمی‌کنیم!
</div>

**Comment by SCR-IR** [comment]
<div dir=rtl>
<b>
درمورد نسخه ۲۴
</b><br/><br/>
سازگاری با گنوم ۴۳ انجام شد.
<br/>
در سال آینده و پس از انتشار نسخه ۱۲ توزیع دبیان، درمورد این افزونه هم إن‌شاء‌الله قصد خانه‌تکانی داریم و علاوه‌بر توقف پیشتیبانی از نسخه‌های قدیمی گنوم، شاید معماری کدمنبع افزونه نیز متحول شود.
<br/><hr/>
* توصیه می‌کنیم همواره از جدیدترین نسخه‌های LTS و توزیع‌های نسبتاً پایدار استفاده بفرمایید. نصب توزیع‌هایی که زودبه‌زود تغییرات اساسی دارند (مثل فدورا) و نسخه‌های آزمایشی بتا و یا نسخه‌های موقّت (مثل اوبونتو 22.10 که LTS نیست) و امثال اینها را هرگز توصیه نمی‌کنیم! 
</div>

**Comment by SCR-IR** [comment]
**درمورد نسخه ۲۷**

یکی از مهم‌ترین تغییرات در این نسخه، به‌کارگیری داده‌های به دست آمده از محاسبات نجومی برای پیش‌بینی رؤیت هلال ماه (ایران) در سال‌های آینده (تا ۱۴۲۰ هـ.ش = ۱۴۶۴ هـ.ق) می‌باشد.
دقت و احتمال صحت داده‌ها برای تعیین ابتدا و انتهای ماه مبارک رمضان در آینده‌ی فوق‌الذکر، بیش از ۸۰ درصد ارزیابی می‌شود.

**Comment by SCR-IR** [comment]
**پشتیبانی از گنوم ۴۵**
از نسخه ۲۹ به بعد افزونه

**Comment by SCR-IR** [comment]
**چالش‌های اخیر و دردسرهای نسخه جدید 🥲**

حتماً متوجه شده بودید که مدتی افزونه را در سایت رسمی گنوم به‌روزرسانی نکردیم و در اینجا نسخه‌های جدید را منتشر می‌کردیم.

اما دلیل آن چه بود؟؟؟

متأسفانه تغییرات زودبه‌زود در سیاست‌های گنوم که هر ۶ ماه یکبار در نسخه‌های اخیر آن وجود داشت، موجب شده برخی کاربران با کتابخانه
Gst
دچار مشکل و اختلال شوند. این کتابخانه برای پخش صداها استفاده میشد که امکانات بسیار متنوعی را ارائه می‌داد.

اما تقریباً از نسخه ۴۶ گنوم تیم پشتیبانی آن با توجه به همین اختلالات مذکور، به استفاده از این کتابخانه در افزونه ما ایراد گرفتند و پیشنهاد دادند برای جلوگیری از اختلالات در آینده، این کتابخانه را استفاده نکنیم.

خب ما موقتاً تا مدتی سعی کردیم با همان ساختار قبلی ادامه بدهیم و نهایتاً امروز مجبور به تغییر ساختار و حذف روش قدیم برای پخش صداها شدیم. چراکه الآن اختلالات یادشده برای تعداد قابل توجهی از کاربران گنوم جدید، پیش آمده بود.

اما چرا مقاومت کردیم و کمی تأخیر داشتیم؟
خب متأسفانه ساختار جدیدی که جایگزین کردیم، امکانات زیادی برای فایل‌های صوتی ندارد و از این پس محدودیت‌هایی خواهیم داشت.
پس جهت اطلاع شما عرض می‌گردد که در نسخه جدید:

۱- تقریباً فقط فایل‌های صوتی با پسوند
ogg
پشتیبانی می‌شوند. پس اگر قبلاً از سایر نوع‌های فایل صوتی استفاده می‌کردید، اکنون باید آن را تبدیل به این قالب کرده و در بخش تنظیمات افزونه نیز جایگزین نمایید.

۲- امکان توقف دستی پخش صداها وجود ندارد و باید تا آخر پخش شوند یا صدای سیستم را دستی ببندید.

۳- در بخش تنظیمات گنوم، بخش صداها، حتماً لازم است تنظیمات مربوط به هشدارهای صوتی را بررسی بفرمایید که غیرفعال نباشد و شدت صدای آن نیز بسته نباشد. توضیحات بیشتر در بخش تنظیمات افزونه درج شده است:

توجه: برای اینکه صداهای اوقات شرعی شنیده شوند، حتماً لازم است در بخش تنظیمات گنوم لینوکس (Settings) ، بخش Sound ، قسمت مربوط به Sounds ، هردو گزینه‌ی Volume Levels و Alert Sound را بررسی نمایید و مطمئن شوید که میزان شدت صدا (در Volume Levels) به مقدار کافی تنظیم شده باشد (ساکت نباشد) و همچنین صدای هشدار (در Alert Sound) نیز غیرفعال (none) نباشد.

**Reply by mimsadAlef** [reply]
یعنی هیچ روشی نیست بشه صدا رو قطع کرد و صدای کل سیستم بسته نشه؟

**Reply by SCR-IR** [reply]
متأسفانه خیر. روشی که روی همه توزیع‌ها بدون مشکل اجرا بشه هنوز پیدا نکردیم
